definitions/features/instance.rb

```diff
@@ -45,7 +45,19 @@ def postgresql_local?
   end
 
   def foreman_proxy_with_content?
-    feature(:foreman_proxy) && feature(:foreman_proxy).with_content? && !feature(:katello)
+    proxy_feature && proxy_feature.with_content? && !feature(:katello)
+  end
+
+  def downstream?
+    feature(:package_manager).satellite_installed? || feature(:package_manager).capsule_installed?
+  end
+
+  def downstream
+    @downstream ||= (feature(:satellite) || feature(:capsule))
+  end
+
+  def proxy_feature
```

Comment:
I'm not very convinced about the concept of the proxy_feature. Wouldn't it be clearer if we have capsule feature as more specific foreman_proxy feature with most of the functionality shared. We would have foreman_proxy feature on both upstream and downstream proxy and capsule feature only on the downstream one. 
This way we could use foreman_proxy if we don't care about up or downstream and test for the capsule feature if we do. We wouldn't need proxy_feature then.

@kgaikwad What do you think?

Reply:
Understood the concern but I thought, having both features foreman_proxy and capsule on downstream will create lot of confusion for others. Every time, we will need to think of when to use what.

Reply:
@mbacovsky @kgaikwad it seems both of you are having valid statements. If we consider @mbacovsky way then in future if we want to share anything between upstream(foreman-proxy) and downstream(Capsule) then this will make it easier. 

As we are still going to have Capsule exclusively for downstream I think we can go with @mbacovsky approach. I am open for more suggestion if any.

Reply:
@kgaikwad my hope was the opposite ;).
While you extracted the common stuff and it is shared among both you can use whatever is present. The rule of thumb should be:
- if you do not care about capsule specific features use `:foreman_proxy` if you are interrested in capsule specifically use `:capsule`.

Does it make things any clearer?

Reply:
After rule of thumb, all my concerns are resolved :-) 

Does that mean now there is no need to create a separate concern file for common things from foreman_proxy & capsule?

Reply:
Thank you @mbacovsky , @upadhyeammit for review.
Done with the code modifications as per suggestions.

Reply:
@kgaikwad, thanks for the update. The code looks way cleaner now. It seems ready to get merged except for one thing I don't understand. The thing is purpose of this `proxy_feature` method. If I'm not mistaken capsule depends on foreman-proxy package therefore `feature(:foreman_proxy)` is present always on upstream and downstream proxy. Actually it is present always as it represents also the internal proxy. So it seems the `proxy_feature` is always true and  equals `feature(:foreman_proxy)`, correct? Or do I miss something?

Also the only usage of the method is probably incorrect. Can we remove this method completely?

Reply:
Decided to refactor this code at the end but I forgot to update :-(
Removed this method completely.